lib_eio_linux/eio_linux.ml

```diff
@@ -1160,11 +1160,87 @@ let net = object
   method getnameinfo = Eio_unix.getnameinfo
 end
 
+let pid_to_process close pid = object
+  inherit Eio.Process.process
+  method pid = pid
+  method status = 
+    match Eio_unix.run_in_systhread @@ fun () -> let v = Unix.waitpid [] pid in close (); v with
+    | _, WEXITED i -> Exited i
+    | _, WSIGNALED i -> Signaled i
+    | _, WSTOPPED i -> Stopped i
+  
+  method stop = 
+    Unix.kill pid Sys.sigkill
+end
+
+let source_or_std fd = function
+  | Some flow -> flow
+  | None -> (source (FD.of_unix_no_hook ~seekable:(FD.is_seekable fd) ~close_unix:true fd) :> Eio.Flow.source)
+
+let sink_or_std fd = function
+  | Some flow -> flow
+  | None -> (sink (FD.of_unix_no_hook ~seekable:(FD.is_seekable fd) ~close_unix:true fd) :> Eio.Flow.sink)
+
+let get_fd_or_err flow =
+  match get_fd_opt flow with
+  | Some fd -> 
+    Unix.dup ~cloexec:false (FD.to_unix `Peek fd)
+  | None -> failwith "Currently only flows backed by file descriptors are supported!"
+
+let open_null flags = lazy (Unix.openfile "/dev/null" flags 0o666)
+
+let dev_null_in = open_null [ Unix.O_RDONLY; Unix.O_CLOEXEC ]
+
+let dev_null_out = open_null [ Unix.O_WRONLY; Unix.O_CLOEXEC ]
+
+let process = object
+  inherit Eio.Process.t
+
+  (* TODO: Is Sys.chdir cwd domain-safe, ? *)
```

Comment:
No, and neither is forcing a promise. You'll probably need some C code for this bit.